django/db/backends/oracle/operations.py

```diff
@@ -399,6 +400,10 @@ def adapt_datetimefield_value(self, value):
         if value is None:
             return None
 
+        # Expression values are handled by the database so are automatically correctly adapted
+        if isinstance(value, Combinable):
```

Comment:
The "right" way to check if something is an expression is by writing `hasattr(value, 'resolve_expression')`. Not all expressions have to be combinables - but they all must be resolvable. There's a few other places this same check is done too, I won't comment on all of them.